### PR TITLE
Ability to count only specific commands, traffic metrics by command types

### PR DESCRIPTION
**Pull request by sergiy0**
AdaptiveAllocationStrategy: Add ability to count only specific commands, add traffic metrics by command types

### CODE REVIEW

File: src/main/scala/com/evolutiongaming/cluster/AdaptiveAllocationStrategy.scala [hunk 1]
```diff
@@ -55,8 +55,11 @@ class AdaptiveAllocationStrategy(
   override def extractShardId(numberOfShards: Int): ShardRegion.ExtractShardId = {
     case x: ClusterMsg =>
       if (!x.isInstanceOf[PersistenceQuery]) {
-        increment(typeName, x.id)
-        metricRegistry.meter(s"persistence.$typeName.sender.${x.id}.$selfHost").mark()
+        if (!x.isInstanceOf[DoNotCount]) {
```
Comment (t3hnar) [inline src/main/scala/com/evolutiongaming/cluster/AdaptiveAllocationStrategy.scala]: let's consider case when we have `Command( Blackjack( PlaceBet | CheckIdle))` how can I disable counting for `CheckIdle` ?

File: src/main/scala/com/evolutiongaming/cluster/AdaptiveAllocationStrategy.scala [hunk 2]
```diff
@@ -55,8 +55,11 @@ class AdaptiveAllocationStrategy(
...
+          increment(typeName, x.id)
+          metricRegistry.meter(s"persistence.$typeName.sender.${ x.id }.$selfHost").mark()
+        }
+        metricRegistry.meter(s"persistence.$typeName.command.${ x.id }.${x.getClass.getCanonicalName}.$selfHost").mark()
```
Comment (t3hnar) [inline src/main/scala/com/evolutiongaming/cluster/AdaptiveAllocationStrategy.scala]: maybe `getSimpleName` is better?

Comment (sergiy0) [inline src/main/scala/com/evolutiongaming/cluster/AdaptiveAllocationStrategy.scala]: Yes, you're right - as we're counting such commands in scope of one persistence entity, there will be no confusion between commands with the same simple name.

File: src/main/scala/com/evolutiongaming/cluster/AdaptiveAllocationStrategy.scala (hunk 1 above)
Comment (sergiy0) [inline src/main/scala/com/evolutiongaming/cluster/AdaptiveAllocationStrategy.scala]: case object CheckIdle extends BlackjackCommand with DoNotCount

Comment (t3hnar) [inline src/main/scala/com/evolutiongaming/cluster/AdaptiveAllocationStrategy.scala]: but it's wrapped in `Command` which does not extends `DoNotCount`

Comment (sergiy0) [inline src/main/scala/com/evolutiongaming/cluster/AdaptiveAllocationStrategy.scala]: `case class Blackjack(value: BlackjackCommand) extends GameCommand with CountControl {
      override def doNotCount: Boolean = value match {
        case CheckIdle => true
        case _                => false
      }
    }`

**Pull request merged**